### ISSUE TITLE
Any way to prevent permanent redirection to https ?

### ISSUE THREAD

**Issue by dylan-brasseur** [nextcloud-snap/nextcloud-snap#2544]
I use my domain for https and http applications, but since nextcloud redirects permanently http requests to https, I need to clear my browser's cache in order to use apps on the same domain but different ports.
I see the permanent redirection in this file : https://github.com/nextcloud-snap/nextcloud-snap/blob/master/src/apache/conf/ssl.conf

Would it be possible (or is it already possible?) to override this redirection to be "temporary" ?

**Comment by scubamuc** [comment]
@dylan-brasseur, not sure if I'm understanding what you're trying to do... usually the request is the other way round see #464

If you're wanting to disable encryption because you're handling certificates alternatively, you can disable enrcyption completely with `sudo nextcloud.disable-https` see [managing encryption](https://github.com/nextcloud-snap/nextcloud-snap/wiki/Managing-encryption)

Not sure what you mean by 'temporary' override though?

**Comment by dylan-brasseur** [comment]
@scubamuc What I mean is allowing to redirect using the 307 http code, instead of the 308 or 301. (as a parameter)

When a website redirects to something else (like http -> https in this case), if the redirection is permanent (301 or 308 http response) then the browser caches the redirection and prevents the user using the nonredirected link (replaces it automatically). 

Let's say you've cleared your browser's cache
Let's say you have nextcloud on **https**://mydomain.com
Let's say you also have some simple nginx server at **http**://mydomain.com:8080

If you type https://mydomain.com, you access nextcloud, if you type **http**://mydomain.com:8080 you access your nginx server, everything's fine
BUT
If by mistake you type  **http**://mydomain.com, then nextcloud will redirect **permanently** to  **https**://mydomain.com
meaning that after that moment, if you type  **http**://mydomain.com:8080, the browser will automatically redirect to  **https**://mydomain.com:8080, which you won't be able to connect to

Then only clearing the cache or using incognito mode will allow you to go to the unsecured server

Basically what I'm asking for is : is it possible to change (maybe with .htaccess ?) or make available as a parameter this line https://github.com/nextcloud-snap/nextcloud-snap/blob/4c6b949918d46809faa5eca9f4f56994625d02a2/src/apache/conf/ssl.conf#L82C88-L82C88 so that the redirect is "temporary"
(R=307 instead of R=permanent)

**Comment by scubamuc** [comment]
@dylan-brasseur, nope, you cannot edit `.htaccess` in Nextcloud snap...

**Comment by Pilzinsel64** [comment]
For me this completely sounds like a browser issue. I mean, I can understand the behavior makes sense in most common cases. But as browser I would let Webserver perform the redirect, if it want to.
However, looks like you need to deal with incognito mode.
The only alternative I could imagine might be a proxy between router and nextckoud. There you can add redirects how you want, but I have no knowledge here. Take a look at our wiki, there are some examples.

**Comment by kyrofa** [comment]
Yeah I'm not sure this is is something we want to support:

1. Changing to a non-permanent redirect for everyone is a performance hit, and generally feels incorrect
2. Adding the ability to configure it requires adding a new knob (something I generally dislike), work to add the feature, as well as work over time to ensure it never breaks

**Comment by dylan-brasseur** [comment]
> Yeah I'm not sure this is is something we want to support, I'm afraid:
> 
> 1. Changing to a non-permanent redirect for everyone is a performance hit, and generally feels incorrect
> 2. Adding the ability to configure it requires adding a new knob (something I generally dislike), work to add the feature, as well as work over time to ensure it never breaks

1 is clearly out of the question, permanent redirects are better in 99.999% of cases for that purpose




> For me this completely sounds like a browser issue. I mean, I can understand the behavior makes sense in most common cases. But as browser I would let Webserver perform the redirect, if it want to. However, looks like you need to deal with incognito mode. The only alternative I could imagine might be a proxy between router and nextckoud. There you can add redirects how you want, but I have no knowledge here. Take a look at our wiki, there are some examples.

Yes, fundamentally it's a browser caching problem. A proxy in front is a solution I think
